include/utils.h

```diff
@@ -0,0 +1,23 @@
+#pragma once
+
+#include <cstdint>
+#include <type_traits>
+
+namespace celerity {
+namespace detail {
+	namespace util {
```

Comment:
Since we already have a `test_utils` (plural) namespace I'd prefer naming this `utils` for consistency.